src/components/main-tools-panel/main-tools-panel.tsx

```diff
@@ -0,0 +1,133 @@
+import { useState } from "react";
+import styled, { css } from "styled-components";
+
+import { darkViolet, primaryViolet } from "../../constants/colors";
+
+import GearIconWhite from "../../../public/icons/gear-white.svg";
+import SettingsIconWhite from "../../../public/icons/settings-white.svg";
+import MemoryIconWhite from "../../../public/icons/memory-white.svg";
+
+import GearIconGrey from "../../../public/icons/gear-grey.svg";
+import SettingsIconGrey from "../../../public/icons/settings-grey.svg";
+import MemoryIconGrey from "../../../public/icons/memory-grey.svg";
+
+import { Layout, Theme } from "../../utils/enums";
+import { useAppLayout } from "../../utils/layout";
+
+enum ActiveButton {
+  options,
+  settings,
+  memory,
+  none,
+}
+
+interface ButtonProps {
```

Comment:
Is there any intention in using `interface` instead of `type`? By design `interface` means the there will be a class created to extend it.

Reply:
Done in all places. You'r right we don't need to use interfaces there.